deepchem/data/datasets.py

```diff
@@ -789,11 +786,13 @@ def iterate(dataset, batch_size):
         else:
           shard_batch_size = batch_size
 
-        num_local_batches = math.ceil(n_shard_samples / shard_batch_size)
-
         if n_shard_samples == 0:
           cur_shard += 1
+          if batch_size is None:
+            cur_global_batch += 1
           continue
+
+        num_local_batches = math.ceil(n_shard_samples / shard_batch_size)
```

Comment:
I modified here due to some special cases in which n_shard_sampels and shard_batch_size are both 0(corresponding to the situation where a dataset have several shards while one shard has no samples in it) Not sure this will cause any further issues.

Reply:
yikes on creating shards with no samples in it.

Reply:
Yep this is super weird, it happens with scaffold splitting in some small datasets.